syntax = "proto3";
import "google/protobuf/wrappers.proto";
import "core/protos/base.proto";

package core.protos.connectors;

message Connector {
  google.protobuf.UInt64Value account_id = 1;
  Source type = 2;
  google.protobuf.StringValue name = 3;
  repeated ConnectorKey keys = 4;
  google.protobuf.UInt64Value id = 5;

}

enum ConnectorMetadataModelType {
  UNKNOWN_MT = 0;
  //New Relic Models
  NEW_RELIC_POLICY = 1;
  NEW_RELIC_CONDITION = 2;
  NEW_RELIC_ENTITY = 3;
  NEW_RELIC_ENTITY_DASHBOARD = 4;
  NEW_RELIC_ENTITY_APPLICATION = 5;
  NEW_RELIC_NRQL = 6;
  NEW_RELIC_ENTITY_DASHBOARD_V2 = 7;

  //Datadog Models
  DATADOG_MONITOR = 101;
  DATADOG_DASHBOARD = 102;
  DATADOG_LIVE_INTEGRATION_AWS = 103;
  DATADOG_LIVE_INTEGRATION_AWS_LOG = 104;
  DATADOG_LIVE_INTEGRATION_AZURE = 105;
  DATADOG_LIVE_INTEGRATION_CLOUDFLARE = 106;
  DATADOG_LIVE_INTEGRATION_FASTLY = 107;
  DATADOG_LIVE_INTEGRATION_GCP = 108;
  DATADOG_LIVE_INTEGRATION_CONFLUENT = 109;
  DATADOG_SERVICE = 110;
  DATADOG_METRIC = 111;
  DATADOG_QUERY = 112;
  DATADOG_APM = 113;

  //Cloudwatch Models
  CLOUDWATCH_METRIC = 201;
  CLOUDWATCH_LOG_GROUP = 202;
  CLOUDWATCH_LOG_GROUP_QUERY = 203;
  CLOUDWATCH_ALARMS = 204;
  RDS_INSTANCES = 205;
  CLOUDWATCH_DASHBOARD = 206;

  //Grafana Models
  GRAFANA_DATASOURCE = 301;
  GRAFANA_DASHBOARD = 302;
  GRAFANA_TARGET_METRIC_PROMQL = 303;
  GRAFANA_PROMETHEUS_DATASOURCE = 304;
  GRAFANA_ALERT_RULE = 305;

  //Clickhouse Models
  CLICKHOUSE_DATABASE = 401;
  CLICKHOUSE_TABLE = 402;

  //Slack Models
  SLACK_CHANNEL = 501;

  //Documentation Models
  MARKDOWN = 601;
  IFRAME = 602;

  //Postgres Models
  POSTGRES_QUERY = 701;
  POSTGRES_TABLE = 702;

  //EKS Models
  EKS_CLUSTER = 801;

  //Sql Database Connection Models
  SQL_DATABASE_CONNECTION_RAW_QUERY = 901;
  SQL_DATABASE_TABLE = 902;

  //Azure Models
  AZURE_WORKSPACE = 1001;
  AZURE_RESOURCE = 1002;

  //Remote Server Models
  SSH_SERVER = 1100;
  SSH_USER = 1101;

  //Mimir Server Models
  GRAFANA_MIMIR_PROMQL = 1201;

  //GKE Models
  GKE_CLUSTER = 1301;

  //MS TEAMS Models
  MS_TEAMS_CHANNEL = 1401;

  //Pagerduty Models
  PAGERDUTY_INCIDENT = 1501;

  //Elastic Search Models
  ELASTIC_SEARCH_INDEX = 1601;
  ELASTIC_SEARCH_SERVICES = 1602;
  ELASTIC_SEARCH_DASHBOARDS = 1603;

  //GCM Models
  GCM_METRIC = 1701;
  GCM_DASHBOARD = 1702;
  GCM_CLOUD_RUN_SERVICE_DASHBOARD = 1703;

  //Zen Duty Models
  ZENDUTY_INCIDENT = 1801;

  //ROOTLY Models
  ROOTLY_INCIDENT = 1901;

  //OPS GENIE Models
  OPS_GENIE_ESCALATION = 2001;
  OPS_GENIE_TEAM = 2002;

  //Google Chat Models
  GOOGLE_CHAT_SPACE = 2101;

  // MongoDB Models
  MONGODB_DATABASE = 2201;
  MONGODB_COLLECTION = 2202;

  //Open Search Models
  OPEN_SEARCH_INDEX = 2301;

  // Asana Models
  ASANA_PROJECT = 2401;

  // Custom Strategies
  MODELS_ACTIVE = 2501;

  //Github Models
  GITHUB_REPOSITORY = 2601;
  GITHUB_MEMBER = 2602;

  // Jira Models
  JIRA_PROJECT = 2701;
  JIRA_USER = 2702;

  // ArgoCd Models
  ARGOCD_APPS = 2801;

  // jenkins Models
  JENKINS_JOBS = 2901;

  // PostHog Models
  POSTHOG_PERSON = 3001;
  POSTHOG_GROUP = 3002;
  POSTHOG_COHORT = 3003;
  POSTHOG_PROPERTY = 3004;

  // ECS Models
  ECS_CLUSTER = 4001;
  ECS_TASK = 4002;
  ECS_SERVICE = 4003;
  
  // SignOz Models
  SIGNOZ_DASHBOARD = 5001;
  SIGNOZ_ALERT = 5002;
  SIGNOZ_METRIC = 5003;

  // Kubernetes Models
  KUBERNETES_NAMESPACE = 6001;
  KUBERNETES_SERVICE = 6002;
  KUBERNETES_DEPLOYMENT = 6003;
  KUBERNETES_INGRESS = 6004;
  KUBERNETES_NETWORK_POLICY = 6005;
  KUBERNETES_HPA = 6006;
  KUBERNETES_REPLICASET = 6007;
  KUBERNETES_STATEFULSET = 6008;

  SENTRY_PROJECT = 7001;

}

message ConnectorKey {
  SourceKeyType key_type = 1;
  google.protobuf.StringValue key = 2;
  google.protobuf.StringValue connector_name = 3;
}
